src/api/error/syntax.rs

```diff
@@ -0,0 +1,44 @@
+// This file is part of rust-web/twig
+//
+// For the copyright and license information, please view the LICENSE
+// file that was distributed with this source code.
+
+use api::Position;
+use api::tokens::{ Token };
+
+#[derive(Debug)]
+pub struct SyntaxError {
+    pub code: SyntaxErrorCode,
+    pub starts_at: Position,
+    pub ends_at: Option<Position>,
+}
```

Comment:
I decided to raise the bar up to Rust level when it comes to error location display :)